class.json-api-endpoints.php

```diff
@@ -1539,12 +1539,23 @@ function current_user_can_access_post_type( $post_type, $context='display' ) {
 
 	function is_post_type_allowed( $post_type ) {
 		// if the post type is empty, that's fine, WordPress will default to post
-		if ( empty( $post_type ) )
+		if ( empty( $post_type ) ) {
 			return true;
+		}
 
 		// allow special 'any' type
-		if ( 'any' == $post_type )
+		if ( 'any' == $post_type ) {
 			return true;
+		}
+
+		if ( $post_type_object = get_post_type_object( $post_type ) ) {
+			if ( ! empty( $post_type_object->show_in_rest ) ) {
+				return true;
+			}
+			if ( ! empty( $post_type_object->publicly_queryable ) ) {
+				return true;
```

Comment:
This should return `$post_type_object->publicly_queryable` in case it's explicitly set to `false`